### PR TITLE
DOC: Clarify draft PR and move from ways to contribute to PR guidelines

### LINKED ISSUE #30436: [Doc]: new contributor guidance on draft PRs
### Documentation Link

https://matplotlib.org/devdocs/devel/contribute.html#ways-to-contribute

### Problem

The paragraph about opening the PR as draft seems out of place under the specific bullet point about being a domain expert.  I’m nit sure where it should go though.

### Suggested improvement

_No response_

### PR DESCRIPTION
**Pull request by timhoffm**
Closes #30436.

### CODE REVIEW

File: doc/devel/pr_guide.rst [hunk 1]
```diff
@@ -12,7 +12,7 @@ We value contributions from people with all levels of experience. In particular,
 if this is your first PR not everything has to be perfect. We'll guide you
 through the PR process. Nevertheless, please try to follow our guidelines as well
 as you can to help make the PR process quick and smooth. If your pull request is
-incomplete or a work-in-progress, please mark it as a `draft pull requests <https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests>`_
+incomplete or a work-in-progress, please mark it as a :ref:`draft pull requests <draft-pr>`
```
Comment (QuLogic) [inline doc/devel/pr_guide.rst]:
```suggestion
incomplete or a work-in-progress, please mark it as a :ref:`draft pull request <draft-pr>`
```

File: doc/devel/pr_guide.rst [hunk 2]
```diff
@@ -118,6 +118,22 @@ Workflow
...
+.. _draft-pr:
+
+Draft PRs
+---------
+
+Authors may create a `draft PR`_ (or change to draft status later) if the code
+is not yet ready for a regular full review. Typical use cases are posting code
+as a basis for discussion or signalling that you intend to rework the code as
+a result of feedback. Authors should clearly communicate why the PR has status
+draft and what needs to be done to make it ready for review. In particular,
```
Comment (QuLogic) [inline doc/devel/pr_guide.rst]:
```suggestion
a result of feedback. Authors should clearly communicate why the PR has draft
status and what needs to be done to make it ready for review. In particular,
```

**Comment by rcomer** [comment]
There is also a paragraph about draft PRs on the development workflow page.  It would be good to link your new explanation from there too.
https://matplotlib.org/devdocs/devel/development_workflow.html#open-a-pull-request

File: doc/devel/pr_guide.rst [hunk 3]
```diff
@@ -118,6 +118,22 @@ Workflow
 Detailed guidelines
 ===================
 
+.. _draft-pr:
```
Comment (story645) [inline doc/devel/pr_guide.rst:121]: Can you update the draft pr link in the new contributor guidance to this link? (GitHub won't let me leave a comment on the exact line)

File: doc/devel/pr_guide.rst [hunk 4]
```diff
@@ -118,6 +118,22 @@ Workflow
...
+
+Draft PRs
+---------
+
+Authors may create a `draft PR`_ (or change to draft status later) if the code
+is not yet ready for a regular full review. Typical use cases are posting code
+as a basis for discussion or signalling that you intend to rework the code as
+a result of feedback. Authors should clearly communicate why the PR has draft
+status and what needs to be done to make it ready for review. In particular,
+they should explicitly ask for specific feedback if needed. By default,
```
Comment (story645) [inline doc/devel/pr_guide.rst]: I think some examples here might be good for expectation setting cause these all seem like 3 rewordings of the same thing to me.

File: doc/devel/pr_guide.rst (hunk 3 above)
Comment (timhoffm) [reply doc/devel/pr_guide.rst:121]: Done.

File: doc/devel/pr_guide.rst (hunk 4 above)
Comment (timhoffm) [reply doc/devel/pr_guide.rst]: I'm reluctant to add examples here. I feel it makes the topic more verbose and does not add much value. These sentences are a nudge on thinking about these related aspects and communicating effectively to reviewers. As indicated by *should*, It's not a hard and detailed requirement that I want or expect users to lengthily craft. Read these sentences and write down what comes to your mind from them. That is good enough.

Here is what AI says - and it very well captures my intention. - I'm using this to illustrate that it's nowadays easily possible for a user to get additional explanation and examples with out us having to spell out everything excessively.

> That guideline is emphasizing **transparency and efficiency** in the pull request (PR) process. Here’s what it really means in practice:
> 
> 1. **Communicate why it’s a draft**
> 
>    * Don’t just mark a PR as *Draft* without context.
>    * Explain whether it’s a work in progress (e.g., missing tests, incomplete implementation, performance concerns, refactor still underway, etc.).
>    * Example: *“Marking this PR as draft because integration tests are failing locally and I need to resolve that before it’s ready.”*
> 
> 2. **Define what’s left to do**
> 
>    * Give reviewers visibility into what’s missing before it can be promoted for full review.
>    * Example: *“Still need to add error handling for edge cases and write unit tests for the new helper functions.”*
> 
> 3. **Request targeted feedback (if needed)**
> 
>    * Even in draft form, you might want feedback on **specific aspects** (design choices, API shape, approach to handling dependencies, etc.).
>    * This helps reviewers focus on what’s most useful at that stage instead of reviewing everything prematurely.
>    * Example: *“At this stage, I’d appreciate feedback on the database schema design before I proceed with writing migrations.”*
> 
> ---
> 
> 👉 In short:
> A draft PR shouldn’t be a mystery. It’s not just “unfinished”—it’s an opportunity to keep collaborators aligned. Clear communication ensures reviewers know **what’s done, what’s not, and where they can help right now.**

File: doc/devel/contribute.rst [hunk 5]
```diff
@@ -287,7 +285,7 @@ guide you through each step:
 4. Check existing pull requests (e.g., :ghpull:`28476`) and filter by the issue number to make sure the issue is not in progress:
 
    * If the issue has a pull request (is in progress), tag the user working on the issue, and ask to collaborate (optional).
-   * If a pull request does not exist, create a `draft pull request <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests>`_ and follow the `pull request guidelines <https://matplotlib.org/devdocs/devel/pr_guide.html>`_.
+   * If a pull request does not exist, create a :ref:`draft pull request <draft-pr>`.
```
Comment (rcomer) [inline doc/devel/contribute.rst]: I had missed this.  My reading of this is that we encourage new contributors to open in draft whether the work is merge-ready or not.

It also doesn’t really make sense to me to go straight from "choose an issue" to "open the [draft] PR".  Maybe this should instead link to "Start a pull request " at the bottom of this page, which links out to the development setup instructions, etc.

Of course, you may consider this comment out-of-scope for the current change.

File: doc/devel/pr_guide.rst (hunk 4 above)
Comment (story645) [reply doc/devel/pr_guide.rst]: Can we link to one of the guides that AI is pulling from? 

> Read these sentences and write down what comes to your mind from them. 

I think I'm getting caught up on what's in scope for specific feedback, and the AI answer of targeted feedback on specific aspects is a bit clearer.

File: doc/devel/contribute.rst (hunk 5 above)
Comment (timhoffm) [reply doc/devel/contribute.rst]: Done.

File: doc/devel/pr_guide.rst (hunk 4 above)
Comment (timhoffm) [reply doc/devel/pr_guide.rst]: Replaced: "specific feedback" -> "targeted feedback"

> Can we link to one of the guides that AI is pulling from?

I don't understand this comment. How do you come to believe AI is pulling from a specific guide? I've just asked to explain my wording in this PR.

**Review by rcomer — APPROVED** [review]
Seems good to me.  Thank you for picking this up @timhoffm.

File: doc/devel/pr_guide.rst (hunk 4 above)
Comment (story645) [reply doc/devel/pr_guide.rst]: > How do you come to believe AI is pulling from a specific guide

Because I'm used to google's version that links to sources. And the writing seemed super cohesive so I was hoping it mostly came from one place.

Comment (timhoffm) [reply doc/devel/pr_guide.rst]: It was ChatGPT :stuck_out_tongue:
